src/sentry/ingest/billing_metrics_consumer.py

```diff
@@ -205,22 +213,45 @@ def _clear_ready_queue(self) -> int:
     def join(self, timeout: Optional[float] = None) -> None:
         deadline = time.time() + timeout if timeout else None
         self._bulk_commit()
+
         while self._ongoing_billing_outcomes:
-            if deadline and deadline > time.time():
+            now = time.time()
+            time_left = now - deadline if deadline else None
```

Comment:
Shouldn't `time_left` be `deadline - now`? Otherwise we pass a negative number to `future.result()`. Then the comparison on line 221 has to be `time_left <= 0`, of course.

Reply:
Good observation!